src/Raven.Client/Documents/Session/AsyncDocumentSession.Stream.cs

```diff
@@ -44,12 +44,14 @@ public void Dispose()
             {
                 while (true)
                 {
+                    var prev = _enumerator.Current;
                     if (await _enumerator.MoveNextAsync().WithCancellation(_token).ConfigureAwait(false) == false)
                         return false;
 
                     _query?.InvokeAfterStreamExecuted(_enumerator.Current);
-
+                    
                     Current = CreateStreamResult(_enumerator.Current);
+                    prev?.Dispose();
```

Comment:
Shouldn't we put here `try / finally` and put the `Dispose` in `finally`?

Reply:
Better to do this in using, yes.

Reply:
This is totally redundant since if something will throw from the session operation it is going to be disposed of thus the context is going to release the memory.